app/models/user.rb

```diff
@@ -3,7 +3,11 @@ class User < ApplicationRecord
   has_many :likes, foreign_key: 'author_id'
   has_many :posts, foreign_key: 'author_id'
 
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true }
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }
+
   def last_three_posts
-    posts.order('created_at desc').limits(3)
+    posts.order('created_at: :desc').limits(3)
```

Comment:
- [ ] You are defining a wrong SQL query here, plus it is dangerous. Kindly refactor to this

  ```
  def last_three_posts
    posts.order(created_at: :desc).limit(3)
  end
  ```
  ![arel-sql](https://user-images.githubusercontent.com/81468675/190830223-3d400b2b-f363-4760-ac6a-880e17c12e37.jpg)